frontend/src/post/PostCard.tsx

```diff
@@ -1,40 +1,70 @@
 import { Link } from "react-router-dom";
 import Post from "./post";
 import { format } from "date-fns";
+import { deletePost } from "./post-service";
+import { useState } from "react";
 
 interface PostCardProps {
   post: Post;
 }
 
 const PostCard: React.FC<PostCardProps> = ({ post }) => {
+  const [deleted, setDeleted] = useState<boolean>(false);
+  const [errorMessage, setErrorMessage] = useState("");
+
+  const handleDelete = async () => {
+    try {
+      if (await deletePost(post.id) == 204) {
```

Comment:
Just do ``await deletePost(post.id); setDeleted(true)``. If no error is thrown, the post is always deleted